### PR TITLE
Newsletters: Add Paywall block settings

### PR DESCRIPTION
**Pull request by lezama**
Add Newsletter settings panel to Paywall Block settings

Screenshot 2023-08-04 at 11 49 34

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* `define( 'JETPACK_BLOCKS_VARIATION', 'beta' );`
* Add a paywall block and make sure Newsletter visibility can be managed from block settings.

### CODE REVIEW

File: projects/plugins/jetpack/extensions/shared/memberships/edit.js [hunk 1]
```diff
@@ -22,24 +20,6 @@ export const useAccessLevel = postType => {
 	return accessLevel;
 };
 
-export const accessOptions = {
```
Comment (simison) [inline projects/plugins/jetpack/extensions/shared/memberships/edit.js:25]: Anything that isn't static feels less of a "constant" to me so access options in this shared file felt better, but not a strong opinion.

Comment (lezama) [reply projects/plugins/jetpack/extensions/shared/memberships/edit.js:25]: you mean is not static because of the translations?

Comment (simison) [reply projects/plugins/jetpack/extensions/shared/memberships/edit.js:25]: Yes, but I'm just being a semantic. :-)

Comment (lezama) [reply projects/plugins/jetpack/extensions/shared/memberships/edit.js:25]: I like semantic too but I need it in a small file so the view bundle doesn't grow

Comment (simison) [reply projects/plugins/jetpack/extensions/shared/memberships/edit.js:25]: You reason is better! 👍 :-)

File: projects/plugins/jetpack/extensions/blocks/paywall/edit.js [hunk 2]
```diff
@@ -29,12 +36,27 @@ function PaywallEdit( { className } ) {
...
+				<span style={ style }>
+					{ text }
+					<Icon icon={ arrowDown } size={ 16 } />
+				</span>
+			</div>
+			<InspectorControls>
+				<PanelBody
+					className="jetpack-subscribe-newsletters-panel"
+					title={ __( 'Newsletter visibility', 'jetpack' ) }
+					icon={ <JetpackLogo showText={ false } height={ 16 } logoColor="#1E1E1E" /> }
```
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/paywall/edit.js]: Let's use the new shared component from https://github.com/Automattic/jetpack/pull/32257

**Review by simison — DISMISSED** [review]
Tests well! 

Just one comment about sharing a component after rebase.

File: projects/plugins/jetpack/extensions/blocks/paywall/edit.js (hunk 2 above)
Comment (lezama) [reply projects/plugins/jetpack/extensions/blocks/paywall/edit.js]: good call

**Review by simison — APPROVED** [review]

**Pull request merged**